### PR TITLE
Added answer choice check

### LINKED ISSUE #20: Validate answer choice when users vote.
Ithere are only two choices and user specifies anything larger or below 0, we accept it with no problems.
But then when you execute /v1/polls/poll_id/result, request hands, because it cannot parse results.

### PR DESCRIPTION
**Pull request by zokir**
closes #20

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -244,13 +249,13 @@ router.post('/polls/:poll_id', function(req, res) {
...
             })
           } else {
-            res.status(404).json({ error: "Not Found", message: "Poll not found"})
+            res.status(404).json({ error: "Not Found", message: "Poll not found"});
           }
       })
     })
 });
 
-router.post('/polls/:poll_id/results', function(req, res) {
```
Comment (armmimic) [inline app.js]: Is this change intentional?

File: app.js [hunk 2]
```diff
@@ -233,8 +233,13 @@ router.post('/polls/:poll_id', function(req, res) {
     .then(() => {
       Poll.findById(parseInt(req.params["poll_id"])).then(poll => {
         if(poll) {
+          const clientAnswer = req.body["answer"];
+          if ((poll["answers"].length <= clientAnswer) || (clientAnswer < 0)) {
```
Comment (armmimic) [inline app.js]: If we're validating, maybe we need to cast `req.body["answer"]` as int as well to do full validation. I have no idea how JS handles int/string comparisons (wut)

File: app.js (hunk 1 above)
Comment (zokir) [reply app.js]: it was, according to spec, but I bet mobile team is not even there eat, so I can switch back. 
tbh I don't know what is grammatically correct here. What do you think

File: app.js (hunk 2 above)
Comment (zokir) [reply app.js]: thanks will cast

File: app.js [hunk 3]
```diff
@@ -390,5 +394,5 @@ router.post('/polls/:poll_id/results', function(req, res) {
 });
 
 app.use('/v1', router);
-app.listen(3000);
-console.log('listening on port ' + 3000);
+app.listen(8081);
```
Comment (armmimic) [inline app.js]: woah this is uncalled for! Maybe make a separate issue and put this into a const at the top?

**Comment by zokir** [comment]
closes #28 created issue as you asked.
